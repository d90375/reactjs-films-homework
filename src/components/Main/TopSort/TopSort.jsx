import React from "react";
import "./TopSort.scss";

const TopSort = () => {
  return (
    <nav className="nav">
      <div className="nav__wrap">
        <div className="nav__title">
          <button type="button" className="nav__title__text nav__title__text_trending">Trending</button>
          <button type="button" className="nav__title__text nav__title__text_rated">Top Rated</button>
          <button type="button" className="nav__title__text nav__title__text_soon">Coming Soon</button>
          <button type="button" className="nav__title__text nav__title__text_genre">Genre</button>
          <div className="nav__title__arrow" />
        </div>
        <div className="nav__format">
          <div className="format__box4">
            <div className="box4 box4_topLeft" />
            <div className="box4 box4_topRight" />
            <div className="box4 box4_btmLeft" />
            <div className="box4 box4_btmRight" />
          </div>
          <div className="format__box2">
            <div className="box2 box2_top" />
            <div className="box2 box2_btm" />
          </div>
        </div>
      </div>
    </nav>
  );
};

export default TopSort;
