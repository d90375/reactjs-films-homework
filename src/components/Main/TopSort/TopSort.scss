.nav {
  border-bottom: 1px solid rgba(36, 43, 45, 0.1);
  margin-bottom: 36px;
}

.nav__title {
  display: flex;
  flex-direction: row;

  .title__text {
    margin-right: 50px;
    color: #0aaee4;
    font-family: "Helvetica Neue", serif;
    font-size: 28px;
    font-weight: 400;
    line-height: 44px;
    //font-weight: 700;
    letter-spacing: 1.12px;

    &:nth-child(4) {
      margin-right: 12px;
    }
  }
}

.title__arrow {
  display: inline-block;
  transform: rotate(45deg);
  margin-top: 14px;
  height: 8px;
  width: 8px;
  border-bottom: 3px solid #0aaee4;
  border-right: 3px solid #0aaee4;
}

.nav__wrap {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 17px;
}

.nav__format {
  display: flex;
  flex-direction: row;
}

.format__box4 {
  display: flex;
  flex-wrap: wrap;
  width: 40px;
  height: 32px;
  margin-right: 16px;
}

.box4 {
  border-radius: 2px;
  width: 16px;
  height: 14px;
  background-color: #3bb4e0;
  opacity: 0.8;
  margin-bottom: 4px;
  margin-right: 4px;

  &:last-child {
    margin-bottom: 0;
  }
}

.box2 {
  width: 35px;
  height: 14px;
  background-color: #242b2d;
  opacity: 0.15;
  border-radius: 2px;

  &:last-child {
    margin-top: 4px;
  }
}
