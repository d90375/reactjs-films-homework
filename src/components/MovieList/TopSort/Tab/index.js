import Tab from "./Tab.container";

export default Tab;
