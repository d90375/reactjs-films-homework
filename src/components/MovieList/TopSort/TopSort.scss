.nav {
  border-bottom: 1px solid rgba(36, 43, 45, 0.1);
  margin-bottom: 2rem;
}

.nav__title {
  display: flex;
  flex-direction: row;

  .nav__title__text {
    padding: 0 10px;
    text-transform: capitalize;
    font-family: "Helvetica Neue", serif;
    font-size: 1rem;
    font-weight: 600;
    line-height: 44px;
    letter-spacing: 1.12px;
    color: #0aaee4;
    background-color: transparent;
    border: none;
    cursor: pointer;
    outline: none;

    &:hover {
      text-shadow: 0 0 10px rgba(10, 174, 228, 1);
    }

    &:nth-child(4) {
      margin-right: 12px;
    }
  }
}

.nav__title__arrow {
  padding: 0 10px;
  margin: 0;
  display: block;
  text-transform: capitalize;
  font-family: "Helvetica Neue", serif;
  font-size: 1rem;
  letter-spacing: 1.12px;
  font-weight: 600;
  line-height: 44px;
  width: 110px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  border: none;
  cursor: pointer;
  background: transparent;
  outline: none;
  color: #0aaee4;

  &:hover {
    text-shadow: 0 0 10px rgba(10, 174, 228, 1);
  }
}

.nav__wrap {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.nav__format {
  display: flex;
  flex-direction: row;
  margin-top: 15px;
}

.format__box4 {
  display: flex;
  flex-wrap: wrap;
  width: 20px;
  height: 16px;
  margin-right: 4px;
}

.box4 {
  border-radius: 2px;
  width: 8px;
  height: 7px;
  background-color: #3bb4e0;
  opacity: 0.8;
  margin-bottom: 2px;
  margin-right: 2px;

  &:last-child {
    margin-bottom: 0;
  }
}

.box2 {
  width: 17px;
  height: 7px;
  background-color: #242b2d;
  opacity: 0.15;
  border-radius: 2px;

  &:last-child {
    margin-top: 2px;
  }
}
