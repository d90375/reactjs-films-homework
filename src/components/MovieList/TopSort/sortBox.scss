.nav__format {
  display: flex;
  flex-direction: row;
  margin-top: 15px;
}

.format__box4 {
  display: flex;
  flex-wrap: wrap;
  height: 16px;
  margin-right: 4px;
  width: 20px;
}

.box4 {
  background-color: #3bb4e0;
  border-radius: 2px;
  height: 7px;
  margin-bottom: 2px;
  margin-right: 2px;
  opacity: 0.8;
  width: 8px;

  &:last-child {
    margin-bottom: 0;
  }
}

.box2 {
  background-color: #242b2d;
  border-radius: 2px;
  height: 7px;
  opacity: 0.15;
  width: 17px;

  &:last-child {
    margin-top: 2px;
  }
}
