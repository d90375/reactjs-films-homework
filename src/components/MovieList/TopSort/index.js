import TopSort from "./TopSort.container";

export default TopSort;
