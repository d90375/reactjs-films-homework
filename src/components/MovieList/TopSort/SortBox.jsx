import React from "react";
import "./sortBox.scss";

const SortBox = () => {
  return (
    <div className="nav__format">
      <div className="format__box4">
        <div className="box4 box4_topLeft" />
        <div className="box4 box4_topRight" />
        <div className="box4 box4_btmLeft" />
        <div className="box4 box4_btmRight" />
      </div>
      <div className="format__box2">
        <div className="box2 box2_top" />
        <div className="box2 box2_btm" />
      </div>
    </div>
  );
};

export default SortBox;
