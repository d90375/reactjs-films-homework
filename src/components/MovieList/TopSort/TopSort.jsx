import PropTypes from "prop-types";
import React from "react";

import Tab from "./Tab";
import SortBox from "./SortBox";

import "./topSort.scss";
import useUrlSearch from "../../../hooks/useURLSearch";

const TopSort = ({
  isDisplayCardDirection,
  handleSwitchToSquare,
  handleSwitchToLine,
  genres,
  onSelectChange,
  handleSelectTab,
  TABS_INFO
}) => {
  const queryParamGenre = useUrlSearch("genreId");
  const selectStyle = queryParamGenre ? `nav__title__arrow nav__active` : `nav__title__arrow`;

  return (
    <nav className="nav">
      <div className="nav__wrap">
        <div className="nav__title">
          {TABS_INFO.map((currTab) => {
            return (
              <Tab key={currTab} onSelectTab={handleSelectTab(currTab)}>
                {currTab}
              </Tab>
            );
          })}
          <select name="genre" className={selectStyle} onChange={onSelectChange}>
            <option value="Genre" hidden>
              Genre
            </option>
            {genres &&
              Object.entries(genres).map(([objKey, genre]) => {
                return (
                  <option key={objKey} value={objKey}>
                    {genre}
                  </option>
                );
              })}
          </select>
        </div>
        <SortBox
          isDisplayCardDirection={isDisplayCardDirection}
          onSwitchToSquare={handleSwitchToSquare}
          onSwitchToLine={handleSwitchToLine}
        />
      </div>
    </nav>
  );
};
export default TopSort;

TopSort.propTypes = {
  genres: PropTypes.objectOf(PropTypes.string),
  TABS_INFO: PropTypes.arrayOf(PropTypes.string),
  onSelectChange: PropTypes.func.isRequired,
  handleSelectTab: PropTypes.func.isRequired
};

TopSort.defaultProps = {
  genres: {},
  TABS_INFO: []
};
