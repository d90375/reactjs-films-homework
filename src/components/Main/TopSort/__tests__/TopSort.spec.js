import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import TopSort from "../TopSort";

const renderer = new ShallowRenderer();
const setUp = (props) => {
  return renderer.render(<TopSort {...props} />);
};

describe("TopSort component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render TopSort component", () => {
    expect(component).toMatchSnapshot();
  });
});
