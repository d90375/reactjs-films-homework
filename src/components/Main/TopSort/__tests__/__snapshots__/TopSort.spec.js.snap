// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSort component should render TopSort component 1`] = `
<div
  className="top"
>
  <div
    className="top__wrap"
  >
    <div
      className="top__title"
    >
      <span
        className="title__text title__text_trending"
      >
        Trending
      </span>
      <span
        className="title__text title__text_rated"
      >
        Top Rated
      </span>
      <span
        className="title__text title__text_soon"
      >
        Coming Soon
      </span>
      <span
        className="title__text title__text_genre"
      >
        Genre
      </span>
      <div
        className="title__arrow"
      />
    </div>
    <div
      className="top__format"
    >
      <div
        className="format__box4"
      >
        <div
          className="box4 box4_topLeft"
        />
        <div
          className="box4 box4_topRight"
        />
        <div
          className="box4 box4_btmLeft"
        />
        <div
          className="box4 box4_btmRight"
        />
      </div>
      <div
        className="format__box2"
      >
        <div
          className="box2 box2_top"
        />
        <div
          className="box2 box2_btm"
        />
      </div>
    </div>
  </div>
</div>
`;
